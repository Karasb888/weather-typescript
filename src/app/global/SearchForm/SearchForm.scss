.textfield {
    background: #e8e8e8;
    padding: 5px 10px;
    border-radius: 11px;
    border-style: none;
    box-sizing: border-box;
    line-height: 35px;
    height: 35px;
    color: #d73c66;
    font-size: 24px;
    outline: none;
    margin-right: 30px;

    &::placeholder {
        color: #5bacbe;
    }
}

.button {
    background: #e8e8e8;
    height: 35px;
    padding: 0 25px;
    line-height: 35px;
    color: #d73c66;
    font-size: 20px;
    border: none;
    outline: none;
    border-radius: 11px;
    cursor: pointer;
    transition: 0.3s;

    &:hover {
        background: #5bacbe;
        color: #d76a3c;
    }
}