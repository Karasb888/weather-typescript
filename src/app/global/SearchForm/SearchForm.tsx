import * as React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { History } from 'history';
import { Button, Grid, TextField } from '@material-ui/core';

interface Props {
    history: History;
};

interface State {
    cityQuery: string;
};

export class SearchForm extends React.Component<Props, State> {
    private input = React.createRef<HTMLInputElement>();

    constructor(props: Props) {
        super(props);

        this.state = {
            cityQuery: ''
        };

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    private handleSubmit(e: React.FormEvent): void {
        e.preventDefault();
        const { cityQuery } = this.state;
        if (cityQuery) {
            this.input.current.value = '';
            this.props.history.push(`/city/${cityQuery}`);
        }
    }

    private handleChange(e: React.ChangeEvent<HTMLInputElement>): void {
        let value = e.target.value;
        value = value.trim();
        this.setState({
            cityQuery: value
        });
    }

    public render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <Grid alignItems="center" container spacing={3}>
                    <Grid xs={9} item>
                        <TextField
                            label="Find weather forecast"
                            placeholder="Type city/country name...(e.g. London)"
                            onChange={this.handleChange}
                            fullWidth
                            inputRef={this.input}
                        />
                    </Grid>
                    <Grid xs={3} item>
                        <Button variant="outlined" color="secondary">
                            Find
                        </Button>
                    </Grid>
                </Grid>
            </form>
        );
    }
};

export default withRouter(connect(null, null)(SearchForm));
