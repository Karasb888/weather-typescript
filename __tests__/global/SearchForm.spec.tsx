import * as React from 'react';
import { shallow } from 'enzyme';
import { SearchForm } from '../../src/app/global/SearchForm/SearchForm';
import { createMemoryHistory } from 'history';

const history = createMemoryHistory();
const props = {
    history
};

describe('++++ <SearchForm /> ++++', () => {
    it('Renders correctly', () => {
        const wrapper = shallow(<SearchForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
