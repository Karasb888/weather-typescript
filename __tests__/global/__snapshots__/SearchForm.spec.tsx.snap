// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`++++ <SearchForm /> ++++ Renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={9}
    >
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        inputRef={
          Object {
            "current": null,
          }
        }
        label="Find weather forecast"
        onChange={[Function]}
        placeholder="Type city/country name...(e.g. London)"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={3}
    >
      <WithStyles(ForwardRef(Button))
        color="secondary"
        variant="outlined"
      >
        Find
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
